k9mail/src/main/java/com/fsck/k9/activity/misc/ContactPictureLoader.java

```diff
@@ -110,59 +111,70 @@ public ContactPictureLoader(Context context, int defaultBackgroundColor) {
 
         mDefaultBackgroundColor = defaultBackgroundColor;
 
-        ActivityManager activityManager =
-                (ActivityManager) appContext.getSystemService(Context.ACTIVITY_SERVICE);
-        int memClass = activityManager.getMemoryClass();
+    }
 
-        // Use 1/16th of the available memory for this memory cache.
-        final int cacheSize = 1024 * 1024 * memClass / 16;
+    public void loadContactPicture(final Address address, final ImageView imageView) {
+        Uri photoUri = mContactsHelper.getPhotoUri(address.getAddress());
+        if (photoUri != null) {
+            final RequestListener<Uri, GlideDrawable> noPhotoListener = new RequestListener<Uri, GlideDrawable>() {
+                @Override
+                public boolean onException(Exception e, Uri model, Target<GlideDrawable> target,
+                        boolean isFirstResource) {
+                    loadFallbackPicture(address, imageView);
+                    return true;
+                }
 
-        mBitmapCache = new LruCache<Address, Bitmap>(cacheSize) {
-            @Override
-            protected int sizeOf(Address key, Bitmap bitmap) {
-                // The cache size will be measured in bytes rather than number of items.
-                return bitmap.getByteCount();
-            }
-        };
+                @Override
+                public boolean onResourceReady(GlideDrawable resource, Uri model,
+                        Target<GlideDrawable> target,
+                        boolean isFromMemoryCache, boolean isFirstResource) {
+                    return false;
+                }
+            };
+
+            Glide.with(imageView.getContext())
+                    .load(photoUri)
+                    .diskCacheStrategy(DiskCacheStrategy.NONE)
+                    .listener(noPhotoListener)
+                    // for some reason, following 2 lines fix loading issues.
+                    .dontAnimate()
+                    .override(mPictureSizeInPx, mPictureSizeInPx)
```

Comment:
I don't fully understand the original role of `mPictureSizeInPx`, but I think this value could be used directly in it.